addon/field-creator.js

```diff
@@ -0,0 +1,1657 @@
+                  })
+                ),
+                React.createElement("td", {style: {padding: "8px", textAlign: "center", borderBottom: "1px solid #ddd"}},
+                  React.createElement("input", {
+                    type: "checkbox",
+                    checked: permissions[name].read,
+                    onChange: () => this.handlePermissionChange(name, "read")
+                  })
+                )
+              )
+            )
+          )
+        )
+      );
+
+    return React.createElement("div", {
+      style: {
+        display: "block",
+        position: "fixed",
+        top: 0,
+        left: 0,
+        width: "100%",
+        height: "100%",
+        backgroundColor: "rgba(0, 0, 0, 0.5)",
+        zIndex: 1050,
+        overflowY: "auto"
+      }
+    },
+    React.createElement("div", {
+      className: "modal-dialog",
+      style: {
+        overflowY: "overlay",
+        position: "absolute",
+        top: "50%",
+        left: "50%",
+        maxHeight: "80%",
+        transform: "translate(-50%, -50%)",
+        maxWidth: "600px",
+        width: "90%",
+        backgroundColor: "#fff",
+        padding: "20px",
+        boxShadow: "0 4px 8px rgba(0, 0, 0, 0.1)",
+        borderRadius: "8px"
+      }
+    },
+    React.createElement("div", {
+      className: "modal-content",
+      style: {
+        position: "relative",
+        height: "100%",
+        display: "flex",
+        flexDirection: "column"
+      }
+    },
+    React.createElement("div", {
+      className: "modal-header",
+      style: {
+        display: "flex",
+        justifyContent: "space-between",
+        alignItems: "center",
+        marginBottom: "15px"
+      }
+    },
+    React.createElement("h1", {className: "modal-title"}, "Set Field Permissions"),
+    React.createElement("button", {
+      type: "button",
+      "aria-label": "Close permission modal button",
+      className: "close",
+      onClick: onClose,
+      style: {
+        cursor: "pointer",
+        background: "none",
+        border: "none",
+        fontSize: "1.5rem",
+        fontWeight: "bold"
+      }
+    }, "×")
+    ),
+    React.createElement("div", {
+      className: "modal-body",
+      style: {
+        overflowY: "auto",
+        flexGrow: 1,
+        marginRight: "-10px", // To compensate for the scrollbar width
+        paddingRight: "10px",
+        scrollbarWidth: "thin",
+        scrollbarColor: "#B0C4DF transparent"
+      }
+    },
+    React.createElement("input", {
+      type: "text",
+      placeholder: "Search profiles and permission sets...",
+      value: this.state.searchTerm,
+      onChange: this.handleSearchChange,
+      style: {
+        width: "100%",
+        padding: "8px",
+        border: "1px solid #ccc",
+        borderRadius: "4px"
+      }
+      
+    }),React.createElement("p", {},"Profile permissions are set to the converted permission set."),
```

Comment:
```suggestion
    }),React.createElement("p", {},"Please consider granting field access to Permission Sets"),
```
It can be great to add this documentation link to explain why users should migrate the fields permissions to PermSets
https://admin.salesforce.com/blog/2022/admin-best-practices-for-user-management